waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -214,28 +215,34 @@
   [restart-expiry-threshold pod]
   (try
     (let [port0 (get-in pod [:spec :containers 0 :ports 0 :containerPort])
-          restart-count (get-in pod [:status :containerStatuses 0 :restartCount])
+          restart-count (get-in pod [:status :containerStatuses 0 :restartCount] 0)
           run-as-user (or (get-in pod [:metadata :labels :waiter/user])
                           ;; falling back to namespace for legacy pods missing the waiter/user label
-                          (k8s-object->namespace pod))]
+                          (k8s-object->namespace pod))
+          {:keys [phase] :as pod-status} (:status pod)
```

Comment:
It would be great to have a comment with a link to the Pod Phase documentation here.
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase

Reply:
Added.